pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -930,6 +933,52 @@ func DeleteXfrm() error {
 	return ee.Error()
 }
 
+// DeleteXfrmWithReqID remove any XFRM policy or state from tables which matches the reqID
+func DeleteXfrmWithReqID(reqID int) error {
```

Comment:
To test this function, you could extend the `TestNodeChurnXFRMLeaks` integration test (with new configuration(s) to cover EncryptedOverlay). It currently exists to ensure that `DeleteXfrm` works as expected and that we don't leak XFRM states or policies.

Reply:
No easy way to test Privileged tests locally, added new test case `TestEncryptedOverlayXFRMLeaks`. Will let CI run it for me.

Reply:
I usually use method 3 from https://docs.cilium.io/en/stable/contributing/testing/unit/#testing-individual-packages, but maybe you mean you're on macOS and it's different there?

Reply:
Running them in linux vm (aarch64) running in mac m2, I think there are some dependencies which are missing from my system. I wish we could create a containerised version of privilege tests. Start container with all dependencies and sandbox the test in it.

Reply:
Actually it works 
`PRIVILEGED_TESTS=true go test -exec "sudo -E" ./pkg/datapath/linux -run=Test/linuxPrivilegedIPv4OnlyTestSuite/TestEncryptedOverlayXFRMLeaks`

It was genuine test failure.